src/coreclr/jit/gentree.h

```diff
@@ -4825,13 +4830,29 @@ struct GenTreeCall final : public GenTree
 
     GenTreeCall(var_types type) : GenTree(GT_CALL, type)
     {
-        fgArgInfo = nullptr;
+        fgArgInfo   = nullptr;
+        gtRetBufArg = nullptr;
     }
 #if DEBUGGABLE_GENTREE
     GenTreeCall() : GenTree()
     {
     }
 #endif
+
+    GenTree* GetLclRetBufArgNode() const
+    {
+        if (gtRetBufArg == nullptr)
+        {
+            return nullptr;
+        }
+
+        assert(HasRetBufArg());
+        return gtRetBufArg->GetNode()->OperIs(GT_ASG) ? gtRetBufArg->GetNode()->gtGetOp2() : gtRetBufArg->GetNode();
+    }
+
+    void SetRetBufArg(Use* retBufArg);
+
+    Use* gtRetBufArg; // The argument that holds return buffer argument
```

Comment:
What is the memory cost of adding this field for SPMI over libs? Growing `GenTreeCall` is pretty costly I believe.

Reply:
(Not strongly opposed to adding it, just curious about the cost)

Reply:
I can measure that. Do you know how to do it though?

Reply:
You can set `s_dspMemStats = true` in compiler.cpp and run superpmi.exe directly over the collection (without parallelism). I believe it should print the stats when it finishes.

Reply:
There are some number I've collected for a similar change here: https://github.com/dotnet/runtime/pull/63402#issuecomment-1006039055, though that's for R2R CoreLib (which should be an upper estimate as I believe R2R tends to have more calls).

Notably, there's some low-hanging fruit in the arg info data structures (padding in `fgArgInfo`, the practically unused `_lateArgInx` in `fgArgTabEntry`) that we could pick to offset the regression.

Reply:
Actually looks like you set it via a COMPlus variable, `COMPlus_JitMemStats=1`.

Reply:
Here are the stats diff: https://www.diffchecker.com/iwAEWDlq. 

Its surprisingly better with this PR, not sure if it has anything to do with skipped marking a node as "address-exposed":

![image](https://user-images.githubusercontent.com/12488060/157184158-8b2dfdda-c4a2-474b-96c9-c76b4d1f5b4a.png)

Reply:
Ah that's good, no worries then! Although I'm curious which collection that is? 5648 methods only seems like few.

Reply:
True. This was `benchmarks`. I will do another round with `libraries_pmi`.

Reply:
https://www.diffchecker.com/q9wrcuLo

![image](https://user-images.githubusercontent.com/12488060/157303634-290b1f9b-bd9f-4d66-a083-69dec67d1707.png)